docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+    "dataSource": "db"
+  }
+}
+
+```
+
+### Applying The category.js Mixin To A Model
+
+To extend the model `Note` with the **category.js** mixin, we need to add a
+**mixins** section in **common/models/note.json** to indicate which mixins
+should be applied to it.
+
+{% include code-caption.html content="common/models/note.json" %}
+
+```json
+{
+  "name": "Note",
+  "properties": {
+    "title": {
+      "type": "string",
+      "required": true
+    },
+    "content": {
+      "type": "string"
+    }
+  },
+  "mixins": {
+    "Category": true
+  }
+}
+```
+
+Specifying a value of **true** for `Category` will apply the **category.js**
+property model mixin to the `Note` model. A value of **false** will not apply
+the mixin.
+
+## LoopBack 4 Approach To Creating A Model Property Mixin
+
+In LoopBack 4, a developer is able to create a model property mixin by:
+
+- placing the mixin class factory function in a separate file
+- generating a model using the CLI as usual
+- adjusting the model file to make use of the mixin class factory function
+
+### Defining The Model Property Mixin Class Factory Function
+
+This mixin class factory function `addCategoryPropertyMixin` in
+**src/mixins/categoryPropertyMixin.ts** adds the required property **category**
+to any model.
+
+{% include code-caption.html content="src/mixins/categoryPropertyMixin.ts" %}
+
+```ts
+import {Constructor} from '@loopback/context';
+
+import {property} from '@loopback/repository';
+
+export function addCategoryPropertyMixin<T extends Constructor<any>>(
+  superClass: T,
+) {
+  class MixedModel extends superClass {
+    @property({
+      type: 'string',
+      required: true,
+    })
+    category: string;
+
+    constructor(...args: any[]) {
+      super(...args);
+    }
+  }
+
+  return MixedModel;
+}
+```
+
+{% include note.html content="At the moment, [TypeScript does not allow decorators in class expressions](https://github.com/microsoft/TypeScript/issues/7342). This is why we need to declare the class with a name, and then return it." %}
+
+### Generating A Model Via The CLI
+
+A CLI-generated model named `Note` with 3 properties: **id**, **title**, and
+**content** would look like this:
+
+{% include code-caption.html content="src/models/note.model.ts" %}
+
+```ts
+import {Entity, model, property} from '@loopback/repository';
+
+@model()
+export class Note extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+    generated: true,
+  })
+  id?: number;
+
+  @property({
+    type: 'string',
+    required: true,
+  })
+  title: string;
+
+  @property({
+    type: 'string',
+  })
+  content?: string;
+
+  constructor(data?: Partial<Note>) {
+    super(data);
+  }
+}
+
+export interface NoteRelations {
+  // describe navigational properties here
+}
+
+export type NoteWithRelations = Note & NoteRelations;
+```
+
+### Adjusting The Model File To Use addCategoryPropertyMixin
+
+Before we use the mixin class factory function:
+
+- rename this class to `TempNote`
```

Comment:
The name `TempNote` suggests to me that this is just a temporary thing that will be removed later.  That's not true, is it?

Let's use a more descriptive class name than `TempNote` please, for example `BaseNote`.

Reply:
> The name TempNote suggests to me that this is just a temporary thing that will be removed  later. That's not true, is it? Let's use a more descriptive class name than TempNote please, for example BaseNote. 

No longer using TempNote